common/iterator/AbstractFunctionalIterator.java

```diff
@@ -220,6 +221,17 @@ public long count() {
         return count;
     }
 
+    @Override
+    public <U> Optional<U> reduce(U initialAccumulator, BiFunction<T, U, U> accumulate) {
```

Comment:
we introduce `reduce()` which can be used to implement list aggregations or summations generically